addons/website/static/src/js/editor/wysiwyg.js

```diff
@@ -164,8 +164,22 @@ Wysiwyg.include({
     _saveElement: async function ($el, context, withLang) {
         var promises = [];
 
-        // Saving a view content
-        await this._super.apply(this, arguments);
+        // Saving Embed Code snippets with <script> in the database, as these
+        // elements are removed in edit mode.
+        if ($el[0].querySelector(".s_embed_code")) {
+            // Copied so as not to impact the actual DOM and prevent scripts
+            // from loading.
+            const clonedEl = $el[0].cloneNode(true);
+            for (const embeddedCodeEl of clonedEl.querySelectorAll(".s_embed_code_embedded")) {
+                if (embeddedCodeEl.dataset.embedContent) {
+                    embeddedCodeEl.innerHTML = embeddedCodeEl.dataset.embedContent;
+                }
+            }
+            await this._super.apply(this, [$(clonedEl), ...[...arguments].slice(1)]);
+        } else {
+            // Saving another view content
```

Comment:
"another" makes this comment strange IMO: you are saving the request view, this is just the case where there is no embed :shrug: